modules/programs/waybar.nix

```diff
@@ -0,0 +1,371 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with lib.types;
+let
+  cfg = config.programs.waybar;
+
+  # Used when generating warnings
+  modulesPath = "programs.waybar.settings.modules";
```

Comment:
I think this should be `programs.waybar.settings[<index>].modules` to match the layout of the options. Maybe that would be part of the mkPath function defined later.

I currently get a bunch of warnings when the `modules` and `modules-<position>` definitions do in fact match.

Reply:
You are right.
There is one additional problem in the generation of warnings, I don't know how to recover the definition number and the entry number correctly.

Right now the warnings are created based on the final configuration (after merging the different definitions).

@rycee do you know how to generate warnings with the appropriate definition entry and index entry (how to access such information)?

Otherwise I will remove the index and definition entry number.

Reply:
I think the placement of the index in the expression is the issue, the warnings I get are like this:

> trace: warning: wkral profile: The module 'clock' defined in 'programs.waybar.settings.modules[definition 1-entry 1]' is not referenced in either `modules-left`, `modules-center` or `modules-right` of Waybar's options

So I think it's just a case of moving the indexing part before the `.modules` reference so it looks like this: `programs.waybar.settings[definition 1-entry 1].modules`. 

Looking a little closer, this change should be made but that's mostly cosmetic so that the path in the warning message matches the path in the config. I found the reason I'm getting the warnings in the first place explained below.